src/Server/DarknetServer.ts

```diff
@@ -0,0 +1,84 @@
+import { DnetServer } from "../DarkNet/models/DarknetServerData";
+import { Icon, labIcon } from "../DarkNet/ui/ServerIcon";
+import { Minigames } from "../DarkNet/controllers/ServerGenerator";
+import { BaseServer, IConstructorParams } from "./BaseServer";
+import { constructorsForReviver, IReviverValue } from "../utils/JSONReviver";
+
+export type IDarknetServer = {
+  /** Hostname. Must be unique */
+  hostname: string;
+  /** IP Address. Must be unique */
+  ip: string;
+  /** Flag indicating whether player has admin/root access to this server */
+  hasAdminRights: boolean;
+  /** Flag indicating whether player is currently connected to this server */
+  isConnectedTo: boolean;
+  /** RAM (GB) used. i.e. unavailable RAM */
+  ramUsed: number;
+  /** RAM (GB) available on this server */
+  maxRam: number;
+  /** Name of company/faction/etc. that this server belongs to, not applicable to all Servers */
+  organizationName: string;
+  /** Flag indicating whether this is a purchased server */
+  purchasedByPlayer: boolean;
+  /** Flag indicating whether this server has a backdoor installed by a player */
+  backdoorInstalled?: boolean;
+  /** If the server has a stasis link applied */
+  hasStasisLink: boolean;
+  /** The amount of ram blocked by the server owner */
+  ramBlock: number;
+  /** The model of the server. Similar models have similar vulnerabilites. */
+  modelId: Minigames;
+  /** The generic password prompt for the server */
+  staticPasswordHint: string;
+  /** Data associated with the password hint */
+  passwordHintData?: string;
+  /** The difficulty rating of the server, associated with its original depth in the net */
+  difficulty: number;
+  /** The depth of the server in the net */
+  x: number;
+  /** The charisma skill required to heartbleed the server */
+  requiredCharismaSkill: number;
+};
+
+export class DarknetServer extends BaseServer implements IDarknetServer, DnetServer {
+  icon: Icon | typeof labIcon;
+  password: string;
+  modelId: Minigames;
+  staticPasswordHint: string;
+  passwordHintData?: string;
+  difficulty: number;
+  x: number;
+  y: number;
+  hasStasisLink: boolean;
+  ramBlock: number;
+  logTrafficInterval: number;
+  requiredCharismaSkill: number;
+
+  constructor(props?: IConstructorParams & DnetServer) {
+    super(props);
+    this.icon = props?.icon || labIcon;
+    this.password = props?.password || "";
+    this.modelId = props?.modelId || "ZeroLogon";
```

Comment:
There doesn't seem like a need for this to be optional; similar to `Server` you can have it take a default value.
```suggestion
  constructor(params: IConstructorParams & DnetServer = {}) {
    super(params);
    this.icon = params.icon ?? labIcon;
    this.password = params.password ?? "";
    this.modelId = params.modelId ?? "ZeroLogon";
...
```